vnext/ReactWindowsCore/tracing/fbsystrace.h

```diff
@@ -51,44 +55,48 @@ class FbSystraceSection {
             .count());
   }
 
-  void init(std::string &&v) {
-    args_[index_++] = std::move(v);
-    begin_section();
+  template <typename... RestArg>
+  FbSystraceSection(uint64_t tag, std::string &&profileName, RestArg &&... rest)
+      : tag_(tag), profile_name_(std::move(profileName)) {
+    id_ = s_id_counter++;
+    init(std::forward<RestArg>(rest)...);
   }
 
-  void init(const std::string &v) {
-    args_[index_++] = v; // copy
-    begin_section();
+  ~FbSystraceSection() {
+    end_section();
   }
 
-  void init() {
-    begin_section();
+ private:
+  template <typename T>
+  using ConsructsToString =
+      std::bool_constant<std::is_convertible<T, std::string>::value || std::is_constructible<std::string, T>::value>;
+
+  template <typename Arg, typename std::enable_if_t<ConsructsToString<Arg>::value> * = nullptr>
+  inline void push_arg(Arg &&arg) {
+    args_[index_++] = std::forward<Arg>(arg);
   }
 
-  template <typename... ConvertsToStringPiece>
-  void init(std::string &&v, ConvertsToStringPiece &&... rest) {
-    args_[index_++] = std::move(v);
-    init(std::forward<ConvertsToStringPiece>(rest)...);
+  template <typename Arg, typename std::enable_if_t<!ConsructsToString<Arg>::value> * = nullptr>
+  inline void push_arg(Arg &&arg) {
+    args_[index_++] = std::to_string(std::forward<Arg>(arg));
   }
 
-  template <typename... ConvertsToStringPiece>
-  void init(const std::string &v, ConvertsToStringPiece &&... rest) {
-    args_[index_++] = v; // copy
-    init(std::forward<ConvertsToStringPiece>(rest)...);
+  void init() {
+    begin_section();
   }
 
-  template <typename... ConvertsToStringPiece>
-  FbSystraceSection(uint64_t tag, std::string &&v, ConvertsToStringPiece &&... rest)
-      : tag_(tag), profile_name_(std::move(v)) {
-    id_ = s_id_counter++;
-    init(std::forward<ConvertsToStringPiece>(rest)...);
+  template <typename Arg>
+  void init(Arg &&arg) {
```

Comment:
>void init(Arg &&arg) { [](start = 2, length = 22)

It seems that this method can be removed without affecting functionality. The two other init() methods must be enough to do the same work,

Reply:
I think you're right, since the base case wll just call the zero rest arg init(). I will try this out.